backend/src/llm/mistral.py

```diff
@@ -9,21 +8,36 @@
 
 
 class Mistral(LLM):
-    client = MistralAsyncClient(api_key=config.mistral_key)
+    client = MistralApi(api_key=config.mistral_key)
 
     async def chat(self, model, system_prompt: str, user_prompt: str, return_json=False) -> str:
         logger.debug("Called llm. Waiting on response model with prompt {0}.".format(str([system_prompt, user_prompt])))
-        response: ChatCompletionResponse = await self.client.chat(
+        response = await self.client.chat.complete_async(
             model=model,
             messages=[
-                ChatMessage(role="system", content=system_prompt),
-                ChatMessage(role="user", content=user_prompt),
+                SystemMessage(content=system_prompt),
+                UserMessage(content=user_prompt),
             ],
             temperature=0,
             response_format={"type": "json_object"} if return_json else None,
         )
+        if response is None:

```

Comment:
Do we need to have separate if checks for Response None and Response.choices?
Won't be easier to just combine these 2,

if not response or not response.choices:
            logger.error("Call to Mistral API failed: No valid response or choices received")
            return "An error occurred while processing the request."

and then we can just proceed with getting the choices

        content = response.choices[0].message.content if response.choices else None
        
        if not content:
            logger.error("Call to Mistral API failed: message content is None or unset")
            return "An error occurred while processing the request."
        // log the content and then just process with returning that content
        logger.debug(f"{model} response: \"{content}\"")
        return content